analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd

```diff
@@ -0,0 +1,169 @@
+
+# Select wanted columns in metadata for merging and assign to a new object
+select_metadata <- metadata %>%
+  dplyr::select(sample_id,
+                Kids_First_Participant_ID,
+                Kids_First_Biospecimen_ID,
+                disease_type_new)
+
+# Read in snv consensus mutation data
+snv_df <-
+  data.table::fread(file.path(root_dir,
+                              "data",
+                              "pbta-snv-consensus-mutation.maf.tsv.gz"))
+```
+
+## Custom Function
+
+```{r}
+# Custom datatable function
+# Function code adapted from: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/49acc98f5ffd86853fc70f220623311e13e3ca9f/analyses/collapse-rnaseq/02-analyze-drops.Rmd#L23
+viewDataTable <- function(data) {
+  DT::datatable(
+    data,
+    rownames = FALSE,
+    filter = "bottom",
+    class = "cell-border stripe",
+    options = list(
+      pageLength = 5,
+      searchHighlight = TRUE,
+      scrollX = TRUE,
+      dom = "tpi",
+      initComplete = htmlwidgets::JS(
+        "function(settings, json) {",
+        "$(this.api().table().header()).css({'background-color':
+                                            '#004467', 'color': '#fff'});",
+        "}"
+      )
+    )
+  )
+}
+```
+
+# Prepare Data 
+
+## SNV consensus mutation data - defining lesions
+
+```{r}
+# Filter the snv consensus mutatation data for the target lesions
+snv_lesions_df <- snv_df %>%
+  dplyr::select(Tumor_Sample_Barcode, Hugo_Symbol, HGVSp_Short) %>%
+  dplyr::mutate(
+    H3F3A.K28M = dplyr::case_when(Hugo_Symbol == "H3F3A" &
+                                    HGVSp_Short == "p.K28M" ~ "Yes",
+                                  TRUE ~ "No"),
+    HIST1H3B.K28M = dplyr::case_when(
+      Hugo_Symbol == "HIST1H3B" & HGVSp_Short == "p.K28M" ~ "Yes",
+      TRUE ~ "No"
+    ),
+    H3F3A.G35R = dplyr::case_when(Hugo_Symbol == "H3F3A" &
+                                    HGVSp_Short == "p.G35R" ~ "Yes",
+                                  TRUE ~ "No"),
+    H3F3A.G35V = dplyr::case_when(Hugo_Symbol == "H3F3A" &
+                                    HGVSp_Short == "p.G35V" ~ "Yes",
+                                  TRUE ~ "No")
+  )
+
+# Join the selected variables from the metadata with the snv consensus mutation
+# and defining lesions data.frame
+snv_lesions_df <- snv_lesions_df %>%
+  dplyr::left_join(select_metadata,
+                   by = c("Tumor_Sample_Barcode" = "Kids_First_Biospecimen_ID")) %>%
+  dplyr::select(
+    Kids_First_Participant_ID,
+    sample_id,
+    Kids_First_Biospecimen_ID = Tumor_Sample_Barcode,
+    dplyr::everything(),
+    -HGVSp_Short,
+    -Hugo_Symbol
+  ) %>%
+  dplyr::distinct() %>%
+  dplyr::mutate(
+    disease_type_reclassified = dplyr::case_when(
+      H3F3A.K28M == "Yes" |
+        HIST1H3B.K28M == "Yes" |
+        H3F3A.G35R == "Yes" |
+        H3F3A.G35V == "Yes" ~ "High-grade glioma",
+      TRUE ~ as.character(disease_type_new)
+    )
+  )
```

Comment:
Can you make the reclassifications further subdivided - something like: 
`HGG, H3 K28 mutant` or `High-grade glioma, H3 K28 mutant`
`HGG, H3 G35 mutant` or `High-grade glioma, H3 G35 mutant`

Reply:
I've made that update in https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/352/commits/a10cdeda7559232acb990392bfe34399c182095f. Note that this table is not the final table from this module, but an interim product https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/352#pullrequestreview-337189596.